File: 1758. minimum-changes-to-make-alternating-binary-string.java
class Solution {
    public int minOperations(String s) {
        int count = 0, n = s.length();
        for (int i = 0; i < n; i++)
            count += (s.charAt(i) ^ i) & 1;
          
        return Math.min(count, n - count);
    }
    
    public static void main(String[] args) {
        Solution sol = new Solution();
        
        // Example 1:
        String s1 = "0100";
        int output1 = sol.minOperations(s1);
        System.out.println("Example 1: Input: s = " + s1);
        System.out.println("Output: " + output1);
        System.out.println("Expected: 1");
        System.out.println();
        
        // Example 2:
        String s2 = "10";
        int output2 = sol.minOperations(s2);
        System.out.println("Example 2: Input: s = " + s2);
        System.out.println("Output: " + output2);
        System.out.println("Expected: 0");
        System.out.println();
        
        // Example 3:
        String s3 = "1111";
        int output3 = sol.minOperations(s3);
        System.out.println("Example 3: Input: s = " + s3);
        System.out.println("Output: " + output3);
        System.out.println("Expected: 2");
    }
}

File: 1758. minimum-changes-to-make-alternating-binary-string.py
class Solution:
    def minOperations(self, s: str) -> int:
        count, n = 0, len(s)
        for i in range(n):
            count += (ord(s[i]) ^ i) & 1
            
        return min(count, n - count)

if __name__ == "__main__":
    sol = Solution()
    
    # Example 1:
    s = "0100"
    output = sol.minOperations(s)
    print(f"Example 1: Input: s = {s}")
    print(f"Output: {output}")
    print(f"Expected: 1")
    print()
    
    # Example 2:
    s = "10"
    output = sol.minOperations(s)
    print(f"Example 2: Input: s = {s}")
    print(f"Output: {output}")
    print(f"Expected: 0")
    print()
    
    # Example 3:
    s = "1111"
    output = sol.minOperations(s)
    print(f"Example 3: Input: s = {s}")
    print(f"Output: {output}")
    print(f"Expected: 2")